### PR TITLE
docs: update 3.6.0 release notes

### PR DESCRIPTION
**Pull request by kayx23**
### Description

Fix a few things in v3.6.0 release notes.

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Review by moonming — APPROVED** [review]

**Review by moonming — DISMISSED** [review]

**Comment by monkeyDluffy6017** [comment]
please resolve the conflicts

**Comment by AlinsRan** [comment]
I think the Chinese content also needs to be modified, such as break change.

File: docs/zh/latest/CHANGELOG.md [hunk 1]
```diff
@@ -79,22 +79,21 @@ title: CHANGELOG
...
 
 ### Core
 
 - :sunrise: 支持配置访问日志的缓冲区大小：[#10225](https://github.com/apache/apisix/pull/10225)
-- :sunrise: 支持在 DNS 发现中传递 resolv.conf：[#9770](https://github.com/apache/apisix/pull/9770)
-- :sunrise: 不再依赖 Rust：[#10121](https://github.com/apache/apisix/pull/10065)
-- :sunrise: 严格验证核心资源的输入：[#10233](https://github.com/apache/apisix/pull/10233)
-- :sunrise: 在 xrpc 中添加 dubbo 协议支持：[#9660](https://github.com/apache/apisix/pull/9660)
+- :sunrise: 支持在 DNS 发现服务中允许配置 `resolv_conf` 来使用本地 DNS 解析器：[#9770](https://github.com/apache/apisix/pull/9770)
+- :sunrise: 安装不再依赖 Rust：[#10121](https://github.com/apache/apisix/pull/10065)
```
Comment (AlinsRan) [inline docs/zh/latest/CHANGELOG.md:88]:
```suggestion
- :sunrise: 安装不再依赖 Rust：[#10121](https://github.com/apache/apisix/pull/10121)
```

**Review by AlinsRan — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]
LGTM

**Pull request merged**